lib/preview/video.js

```diff
@@ -0,0 +1,146 @@
+ * notice, this paragraph and the following two paragraphs appear in all copies,
+ * modifications, and distributions.
+ *
+ * Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var exec = require('child_process').exec;
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var PreviewConstants = require('./constants');
+var PreviewImage = require('./image');
+var Result = require('./model').Result;
+
+/**
+ * Process a video file
+ *
+ * @param  {Context}    ctx                 The job context
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error object, if any
+ * @param  {Result}     callback.result     The result for the video file
+ */
+var process = module.exports.process = function(ctx, callback) {
+  extractFirstFrame(ctx, function(err, imagePath) {
+    if (err) {
+      return callback(err);
+    }
+
+    PreviewImage.processImage(ctx, imagePath, function(err, imageResult) {
+      if (err) {
+        return callback(err);
+      }
+
+      convertVideoIfNecessary(ctx, function(err, convertedVideoPath) {
+        if (err) {
+          return callback(err);
+        }
+
+        if (convertedVideoPath) {
+          imageResult.metadata.converted_video = convertedVideoPath;
+        }
+        var result = new Result(PreviewConstants.STATUS.DONE, imageResult.thumbnail, imageResult.image, null, imageResult.metadata);
+
+        return callback(null, result);
+      });
+    });
+  });
+};
+
+/**
+ * Extract the first frame as a PNG image
+ *
+ * @param  {Context}    ctx                          The job context
+ * @param  {Function}   callback                     Standard callback function
+ * @param  {Object}     callback.err                 An error object, if any
+ * @param  {String}     callback.imagePath           The path to the extracted image
+ */
+var extractFirstFrame = function(ctx, callback) {
+  var extractCommand = [
+    'ffmpeg',
+    util.format('-i "%s"', ctx.sourcePath),
+    '-vframes 1',
+    path.join(ctx.directory, 'frame%d.png')
```

Comment:
@johncrossman Frame number; for some reason ffmpeg requires this format even if it's only outputting one frame. `frame1.png` (see below) is the result.